Formula/neomutt.rb

```diff
@@ -0,0 +1,72 @@
+class Neomutt < Formula
+  desc "Teaching an Old Dog New Tricks"
+  homepage "https://www.neomutt.org/"
+  url "https://github.com/neomutt/neomutt.git", :tag => "neomutt-20171215", :revision => "ae61285170533b4be544e738cdd2eedefd1856ff"
+  head "https://github.com/neomutt/neomutt.git", :branch => "master"
+
+  option "with-lmdb", "Build with lmdb support"
+  option "with-lua", "Build with lua scripting support enabled"
+  option "with-s-lang", "Build against slang instead of ncurses"
+
+  # Neomutt-specific patches
+  option "with-notmuch-patch", "Apply notmuch patch"
+
+  depends_on "gettext" => :build
+  depends_on "docbook-xsl" => :build
+  depends_on "libxslt" => :build unless OS.mac?
+  depends_on "krb5" => :build unless OS.mac?
+
+  depends_on "openssl"
+  depends_on "tokyo-cabinet" => :recommended if build.without?("lmdb")
+  depends_on "lmdb" => :optional
+  depends_on "gpgme" => :optional
+  depends_on "libidn" => :optional
+  depends_on "lua" => :optional
+  depends_on "s-lang" => :optional
+  depends_on "notmuch" if build.with? "notmuch-patch"
+
+  def install
+    # Find our docbook catalog
+    ENV["XML_CATALOG_FILES"] = "#{etc}/xml/catalog"
+
+    args = %W[
+      --prefix=#{prefix}
+      --with-ssl=#{Formula["openssl"].opt_prefix}
+      --gss
+      --disable-idn
+    ]
+
+    args << "--enable-gpgme" if build.with? "gpgme"
+
+    # Neomutt-specific patches
+    args << "--notmuch" if build.with? "notmuch-patch"
+
+    args << "--sasl" unless OS.linux?
+
+    if build.with? "lmdb"
+      args << "--lmdb"
+    else
+      args << "--tokyocabinet"
+    end
+
+    if build.with? "lua"
+      args << "--lua"
+      args << "--with-lua=#{Formula["lua"].prefix}"
+    end
+
+    if build.with? "s-lang"
+      args << "--with-ui=slang"
+    else
+      args << "--with-ui=ncurses"
+    end
+
+    system "./configure", *args
+    system "make"
+
+    system "make", "install"
+  end
+
+  test do
+    system bin/"neomutt", "-D"
```

Comment:
The test should exercise and verify the functionality of the software. See `dep`, `jose`, `faas-cli`, and `s-nail` for some good examples.

Reply:
@jindraj 
Our `make test` target has a few unit tests.

Reply:
`make test` is only permitted for security software like openssl. In the `test do` block, the build environment is long gone.

Reply:
@ilovezfs Is something wrong with `neomutt -D` or `neomutt -v`? Actually I don't have any kind of test like that on my mind. Shouldn't be sufficient to see if binary is compiled and executable? I mean it executes and exits with zero exit code.

Reply:
Can we check the presence of some "standard" configuration variable in the output of `neomutt -D`?

Reply:
@fxcoudert We _could_ do.
By default `neomutt -D` will try to read a system and a user config file.
They won't exist in the build environment, so `-D` will just show the defaults.

There are ~400 config variables in the current version
(depending on which options are configured) and the output isn't sorted.

I suggest: `neomutt -n -F /dev/null -D | grep "^index_format="`

(`-n -F /dev/null` tells neomutt to **not** read any config files)

It's not much of a test.

@jindraj Please could you mind make the changes.

Thanks.